### ISSUE TITLE
MPI_Neighbor_alltoallv_init() should not necessarily error on null arguments

### ISSUE THREAD

**Issue by jczhang07** [open-mpi/ompi#12037]
In OMPI-5.0, `MPI_Neighbor_alltoallv_init()` errors with NULL counts and displs arguments, which actually could happen when some ranks have zero indegree or outdegree. 

https://github.com/open-mpi/ompi/blob/5cb309412872389a54f074ab867eb9f8d6267a9e/ompi/mpi/c/neighbor_alltoallv_init.c#L105

This bug caused an error in petsc

```
[5]PETSC ERROR: General MPI error
[5]PETSC ERROR: MPI error 13 MPI_ERR_ARG: invalid argument of some other kind
```

**Comment by ggouaillardet** [comment]
Thanks for the report!

I issued #12039 to fix this.
Meanwhile, you can manually download and apply the patch at https://github.com/open-mpi/ompi/pull/12039.patch

**Comment by prj-** [comment]
@ggouaillardet, this issue should be closed, I can confirm it is fixed in Open MPI 5.0.1.
